syntax = "proto3";
package user;
import "google/protobuf/timestamp.proto";

message CreateUserRequest {
    string userId = 1;
    string nickName = 2;
    string avatar = 3;
    google.protobuf.Timestamp createdAt = 4;
    bool isBanned = 5;
    google.protobuf.Timestamp banExpirationDate = 6;
    string bannedByAdminId = 7;
}

message GetUserRequest {
    string userId = 1;
}

message BanUnbanPlayerRequest {
    string userId = 1;
    bool isBanned = 2;
    google.protobuf.Timestamp banExpirationDate = 3;
}

message GetUserResponse {
    //TODO 

    //this is mock
    string userId = 1;
    string nickName = 2;
    string avatar = 3;
    google.protobuf.Timestamp createdAt = 4;
    bool isBanned = 5;
    google.protobuf.Timestamp banExpirationDate = 6;
    string bannedByAdminId = 7;
}

message UpdateAvatarRequest {
    string userId = 1;
    string avatar = 2;
}

message InfoResponse {
    string info = 1;
}

service UserService {
    rpc CreateUser(CreateUserRequest) returns (InfoResponse) {}
    rpc GetUser(GetUserRequest) returns (GetUserResponse) {}
    rpc UpdateAvatar(UpdateAvatarRequest) returns (InfoResponse) {}
    rpc BanUnbanPlayer(BanUnbanPlayerRequest) returns (InfoResponse) {}
}